bundles/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeDimmerHandler.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        updateDeviceId();
+    }
+
+    public void updateDimmerOutputIntensity(int outputIntensity) {
+        updateState(SinopeBindingConstants.CHANNEL_DIMMER_OUTPUTINTENSITY, new PercentType(outputIntensity));
+    }
+
+    public void updateLightMode(int lightMode) {
+        updateState(SinopeBindingConstants.CHANNEL_LIGHTMODE, new DecimalType(lightMode));
+    }
+
+    public void update() throws IOException {
+        if (this.deviceId != null) {
+            if (isLinked(SinopeBindingConstants.CHANNEL_DIMMER_OUTPUTINTENSITY)) {
+                this.updateDimmerOutputIntensity(readOutputIntensity());
+            }
+            if (isLinked(SinopeBindingConstants.CHANNEL_LIGHTMODE)) {
+                this.updateLightMode(readLightMode());
+            }
+
+        } else {
+            logger.error("Device id is null for Thing UID: {}", getThing().getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    private int readOutputIntensity() throws UnknownHostException, IOException {
+        SinopeDataReadRequest req = new SinopeDataReadRequest(this.getSinopeGatewayHandler().newSeq(), deviceId,
+                new SinopeOutputIntensityData());
+        logger.debug("Reading Output Intensity for device id: {}", ByteUtil.toString(deviceId));
+        SinopeDataAnswer answ = (SinopeDataAnswer) this.getSinopeGatewayHandler().execute(req);
+        int intensity = ((SinopeOutputIntensityData) answ.getAppData()).getOutputIntensity();
+        logger.debug("Output intensity is : {} %", intensity);
+        return intensity;
+    }
+
+    private int readLightMode() throws UnknownHostException, IOException {
+        SinopeDataReadRequest req = new SinopeDataReadRequest(this.getSinopeGatewayHandler().newSeq(), deviceId,
+                new SinopeLightModeData());
+        logger.debug("Reading Light Mode for device id: {}", ByteUtil.toString(deviceId));
+        SinopeDataAnswer answ = (SinopeDataAnswer) this.getSinopeGatewayHandler().execute(req);
+        int lightMode = ((SinopeLightModeData) answ.getAppData()).getLightMode();
+        logger.debug("Light mode is : {}", lightMode);
+        return lightMode;
+    }
+
+    private void updateDeviceId() {
+        String sDeviceId = (String) getConfig().get(SinopeBindingConstants.CONFIG_PROPERTY_DEVICE_ID);
+        this.deviceId = SinopeConfig.convert(sDeviceId);
+        if (this.deviceId == null) {
+            logger.debug("Invalid Device id, cannot convert id: {}", sDeviceId);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid Device id");
+            return;
+        }
+        Bridge bridge = this.getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
```

Comment:
I believe this should be **CONFIGURATION_ERROR** is no bridge was configured. Please see #12331.